spec/lang/step.md

```diff
@@ -535,26 +550,26 @@ impl<M: Memory> Machine<M> {
         );
         let func = frame.func;
 
-        let Some(caller_return_info) = frame.caller_return_info else {
-            // Only the bottom frame in a stack has no caller.
-            // Therefore the thread must terminate now.
-            assert_eq!(Int::ZERO, self.thread_manager.active_thread().stack.len());
-
-            return self.thread_manager.terminate_active_thread();
```

Comment:
Ah, this is a good catch, currently we are failing to free the locals of the bottom frame.

We should probably have some tests that check for memory leaks...